docs/migrating-v1alpha1.Run-to-v1beta1.CustomRun.md

```diff
@@ -0,0 +1,194 @@
+kind: CustomRun
+metadata:
+  name: customrun-with-spec
+spec:
+  customSpec:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    spec:
+      field1: value1
+      field2: value2
+```
+
+### Changes to the reference
+
+In `v1beta1.CustomRun`, the specification is renamed from `spec.ref` to `spec.customRef`.
+
+```yaml
+# before (v1alpha1.Run)
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  name: run-with-reference
+spec:
+  ref:
+    apiVersion: example.dev/v1alpha1
+    kind: Example
+    name: my-example
+---
+# after (v1beta1.CustomRun)
+apiVersion: tekton.dev/v1beta1
+kind: CustomRun
+metadata:
+  name: customrun-with-reference
+spec:
+  customRef:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    name: my-customtask
+```
+
+### Support `PodTemplate` in Custom Task Spec
+
+`spec.podTemplate` is removed in `v1beta1.CustomRun`. You can support that field in your own custom task spec if you want to. For example:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: CustomRun
+metadata:
+  name: customrun-with-podtemplate
+spec:
+  customSpec:
+    apiVersion: example.dev/v1beta1
+    kind: Example
+    spec:
+      podTemplate: 
+        securityContext:
+        runAsUser: 1001
+```
+
+## Changes to implementation instructions
+We've changed four implementation instructions. Note that `status reporting` is the only required instruction to follow, others are recommendations.
+
+### Status Reporting
+
+You **MUST** report `v1beta1.CustomRun` as `Done` (set its status as `True` or `False`) **ONLY** when you want Pipeline Controller consider it as finished.
+
+For example, if the `CustomRun` failed, while it still has remaining `retries`. If you want pipeline controller to continue watching its status changing, you **MUST NO** mark it as `Done`. Otherwise, the `PipelineRun` controller may consider it as finished.
+
+Here are statuses that indicating the `CustomRun` is done.
+```yaml
+Type: Succeeded
+Status: False
+Reason: TimedOut
+```
+```yaml
+Type: Succeeded
+Status: True
+Reason: Succeeded
+```
+
+### Cancellation
+
+Custom Task implementors are responsible for implementing `cancellation` to **support pipelineRun level timeouts and cancellation**. If a Custom Task implementor does not support cancellation via `customRun.spec.status`, `PipelineRun` can not timeout within the specified interval/duration and can not be cancelled as expected upon request.
+
+It is recommended to update the `CustomRun` status  as following, once noticing `customRun.Spec.Status` is updated to `RunCancelled`
+
+```yaml
+Type: Succeeded
+Status: False
+Reason: CustomRunCancelled
+```
+
+### Timeout
+
+We recommend `customRun.TimeOut` to be set for each retry attempt instead of all retries. That's said, if one `CustomRun` execution fails on `Timeout` and it has remaining retries, the `CustomRun` controller SHOULD NOT set the status of that `CustomRun` as `False`. Instead, it SHOULD initiate another round of execution.
```

Comment:
```suggestion
We recommend `customRun.Timeout` to be set for each retry attempt instead of all retries. That said, if one `CustomRun` execution fails on `Timeout` and it has remaining retries, the `CustomRun` controller SHOULD NOT set the status of that `CustomRun` as `False`. Instead, it SHOULD initiate another round of execution.
```